dbms/src/Functions/FunctionsConversion.h

```diff
@@ -471,15 +547,22 @@ struct ConvertThroughParsing
     static void execute(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count,
                         Additions additions [[maybe_unused]] = Additions())
     {
-        using ColVecTo = std::conditional_t<IsDecimalNumber<ToFieldType>, ColumnDecimal<ToFieldType>, ColumnVector<ToFieldType>>;
+        using ColVecTo = typename ToDataType::ColumnType;
 
         const DateLUTImpl * local_time_zone [[maybe_unused]] = nullptr;
         const DateLUTImpl * utc_time_zone [[maybe_unused]] = nullptr;
 
         /// For conversion to DateTime type, second argument with time zone could be specified.
-        if constexpr (std::is_same_v<ToDataType, DataTypeDateTime>)
+        if constexpr (std::is_same_v<ToDataType, DataTypeDateTime> || std::is_same_v<ToDataType, DataTypeDateTime64>)
```

Comment:
Could we extrace std::is_same_v<FromDataType, DataTypeDateTime64> as named constexpr bool into class context?

Reply:
fixed